actionpack/lib/action_dispatch/http/upload.rb

```diff
@@ -41,8 +41,8 @@ def initialize(hash) # :nodoc:
           @original_filename = nil
         end
 
-        @content_type      = hash[:type]
-        @headers           = hash[:head]
+        @content_type = hash[:type] || hash[:content_type]
+        @headers      = hash[:head] || hash[:headers]
```

Comment:
What do you think about deprecating the old, unintuitive ones?

Reply:
Might also be nice to switch to keyword arguments while we're here. Saves us the ArgumentError when missing `:tempfile`.

Reply:
> What would the process be for creating the deprecation? Is there a guide?

Should be as simple as:

```ruby
# before line 44
if hash.key?(:type)
  ActiveSupport::Deprecation.warn("...")
end

if hash.key?(:head)
  # ...
```

Note that you'll need to add `require "active_support/deprecation` on the top of the file too. You can try grep the project for `Deprecation.warn` to see examples.

---

> what version should we target for deprecation?

I think it's safe to say _this will be removed in the next major version of Rails_, which essentially mean 6.0

---

And yes, let's change to kwargs when we're at it. 💯